### ISSUE TITLE
[Feature Request] Add workflow task failure policy worker option

### ISSUE THREAD

**Issue by cretz** [temporalio/sdk-core#571]
### Describe the solution you'd like

Go has a `worker.Options.WorkflowPanicPolicy` which defaults to `BlockWorkflow` but you can set it to `FailWorkflow`. Not sure we need that exact terminology, but we should have a worker option that automatically promotes an activation completion failure to a workflow failure.

**Comment by Sushisource** [comment]
We have this now with the failure types, yes? Closing b/c of that. We can re-open if not.

**Comment by cretz** [comment]
Yes, we already have basically this in SDKs now (an the one core-level task failure, non-determinism error, we do allow you to customize whether it's a task failure or not)
